tracer/src/Datadog.Trace/Agent/AgentWriter.cs

```diff
@@ -135,6 +140,48 @@ public void WriteTrace(ArraySegment<Span> trace)
             }
         }
 
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private void LogSpanDebugInformation(TraceContext traceContext, ArraySegment<Span> trace)
+        {
+            foreach (var span in trace)
+            {
+                Log.Debug(
+                    "Span write to agent: [s_id: {SpanId}, p_id: {ParentId}, t_id: {TraceId}] for (Service: {ServiceName}, Resource: {ResourceName}, Operation: {OperationName}, SamplingPriority: {SamplingPriority} Tags: [{Tags}])\nDetails:{ToString}",
+                    new object[] { span.Context.RawSpanId, span.Context.ParentIdInternal, span.Context.RawTraceId, span.ServiceName, span.ResourceName, span.OperationName, traceContext.SamplingPriority, span.Tags, span.ToString() });
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private void LogRootSpanDebugInformation(TraceContext traceContext, Span span)
+        {
+            Log.Debug(
+                    "RootSpan write to agent: [s_id: {SpanId}, p_id: {ParentId}, t_id: {TraceId}] for (Service: {ServiceName}, Resource: {ResourceName}, Operation: {OperationName}, SamplingPriority: {SamplingPriority} Tags: [{Tags}], TraceContext Tags: [{TraceTags}])\nDetails:{ToString}",
+                    new object[] { span.Context.RawSpanId, span.Context.ParentIdInternal, span.Context.RawTraceId, span.ServiceName, span.ResourceName, span.OperationName, traceContext.SamplingPriority, span.Tags, traceContext.Tags, span.ToString() });
+        }
+
+        private string GetSpanInfoForDebugLog(Span span)
+        {
+            var sb = StringBuilderCache.Acquire(StringBuilderCache.MaxBuilderSize);
+            sb.AppendLine($"TraceId64: {span.Context.TraceId128.Lower}");
+            sb.AppendLine($"TraceId128: {span.Context.TraceId128}");
+            sb.AppendLine($"RawTraceId: {span.Context.RawTraceId}");
+            sb.AppendLine($"ParentId: {span.Context.ParentIdInternal}");
+            sb.AppendLine($"SpanId: {span.Context.SpanId}");
+            sb.AppendLine($"RawSpanId: {span.Context.RawSpanId}");
+            sb.AppendLine($"Origin: {span.Context.Origin}");
+            sb.AppendLine($"ServiceName: {span.ServiceName}");
+            sb.AppendLine($"OperationName: {span.OperationName}");
+            sb.AppendLine($"Resource: {span.ResourceName}");
+            sb.AppendLine($"Type: {span.Type}");
+            sb.AppendLine($"Start: {span.StartTime.ToString("O")}");
+            sb.AppendLine($"Duration: {span.Duration}");
+            sb.AppendLine($"End: {span.StartTime.Add(span.Duration).ToString("O")}");
+            sb.AppendLine($"Error: {span.Error}");
+            sb.AppendLine($"Meta: {span.Tags}");
```

Comment:
If we include the string tags (`meta`), we should probably include the numeric tags (`metrics`) as well.

Reply:
I removed this, but looking at this it does include the metrics as it logs out all of the `meta` and `metric` data in the Tags

So maybe a better name instead of `Meta` would just be `Tags`?

Reply:
Agreed. Thank you!